src/main/java/uk/gov/hmcts/dm/controller/StoredDocumentController.java

```diff
@@ -99,11 +100,18 @@ private void init() throws Exception {
 
     @DeleteMapping(value = "{id}")
     @ApiOperation("(Soft) Deletes a Stored Document.")
-    @ApiResponses(value={
-        @ApiResponse(code=405, message = "Method not implemented at the moment")
-    })
+
     public ResponseEntity<Object> delete(@PathVariable UUID id) {
-        return ResponseEntity.status(405).build();
+        auditedStoredDocumentOperationsService.deleteStoredDocument(id);
+        return ResponseEntity.status(HttpStatus.GONE).build();
+    }
+
+    @DeleteMapping(value = "{id}/removePermanently")
+    @ApiOperation("Hard deletes a Stored Document.")
+
+    public ResponseEntity<Object> removePermanently(@PathVariable UUID id) {
```

Comment:
Perhaps information on whether the document should be deleted permanently onr not could be included in query param in a single url (defaulting to 'false' I guess)?

ie:

```
DELETE /documents/123&permanently=true`
DELETE /documents/123
```

https://hmcts.github.io/restful-api-standards/#141